### ISSUE TITLE
Got 10 doesn't work with AWS XRay in lambda

### ISSUE THREAD

**Issue by MickVanDuijn** [aws/aws-xray-sdk-node#240]
#### Describe the bug

 - Node.js version: 10.x and 12.x
 - OS & version: AWS Lambda, Got 10

This occurs when using Got 10 in combination with AWS XRay on AWS Lambda. **Note**: Got 9 works fine with the provided snippet. 

I also opened an issue on the [`got`](https://github.com/sindresorhus/got/issues/1022) repository.

#### Actual behavior

Whenever XRay capture is enabled, the lambda doesn't return any response, and the actual request is made against `https://localhost`. Subsequent executions take longer and longer (even up to 2 minutes) until finally one execution will show a bunch of LAMBDA_RUNTIME_ERROR. 

**Edit**: the lambda execution actually dies midway, while without XRay enabled, it successfully completes.

#### Expected behavior

Get contents of `google.com` and return the specified response (`body: 'ok'`)

#### Code to reproduce

```js
require('aws-xray-sdk-core').captureHTTPsGlobal(require('https'))
const got = require('got')

export async function handler() {
    await got.get(`https://google.com`)

    return {
        statusCode: 200,
        body: 'ok',
    }
}
```

<img width="1171" alt="image" src="https://user-images.githubusercontent.com/1350647/72446204-6320f780-37b3-11ea-956e-fc53477c3d1e.png">

**Comment by MickVanDuijn** [comment]
@chanchiem should this also receive the `bug` label? It is killing my executions midway, while it worked fine without XRay enabled

**Comment by willarmiros** [comment]
Hi @MickVanDuijn,
Thank you for raising this issue. I was able to reproduce it and the specific fault I got on the X-Ray console was:
```
Error: connect ECONNREFUSED 127.0.0.1:443
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1128)
```

That makes me think this issue might be related to the one in #217, and it seems like that host implies the HTTP request can't even get out to the internet. I'm investigating that issue with Lambda, and I'll add this issue to that investigation.

**Comment by MickVanDuijn** [comment]
Thanks @willarmiros ! What's also strange is that the requested URL turned into `localhost`, I don't see that on #217

**Comment by willarmiros** [comment]
Hi @MickVanDuijn,
I believe I've found the root cause of the issue. The `got` library overwrites existing headers in requests it makes, however the X-Ray SDK requires its specific header to be attached to all outgoing HTTP requests in order for it to trace them. You can read about the structure of the header [here](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader). You can get your header by using the X-Ray SDK's APIs, or you *should* be able to extract it from the `event.headers` object.

As for forcing the header into the `got` request, you'll have to look into their documentation for that. I hope this helps and please let me know if this resolves your issue!

**Comment by MickVanDuijn** [comment]
As @szmarczak stated (https://github.com/sindresorhus/got/issues/1022), the `http/https` patcher doesn't support the newer versions of the `https.request` syntax. I created #247 to fix that :)

**Comment by willarmiros** [comment]
Thanks @MickVanDuijn! I'll follow up on the PR.

**Comment by willarmiros** [comment]
Fix released in v3.0.0-alpha.2
